src/components/popup.tsx

```diff
@@ -42,151 +32,78 @@ export interface PopupProps {
 }
 
 export interface PopupState {
-    inputValue: string;
-    previousInputValue: string;
-    validationErrorMsg: string;
-    primaryBtnSpinner: boolean;
-    secondaryBtnSpinner: boolean;
+    primaryBtnShowSpinner: boolean;
+    secondaryBtnShowSpinner: boolean;
 }
 
 export class Popup extends React.Component<PopupProps, PopupState> {
 
     state = {
-        inputValue: undefined,
-        previousInputValue: undefined,
-        validationErrorMsg: undefined,
-        primaryBtnSpinner: false,
-        secondaryBtnSpinner: false,
+        primaryBtnShowSpinner: false,
+        secondaryBtnShowSpinner: false,
     }
 
-    private popupBodyRef: React.RefObject<HTMLDivElement>;;
-
     constructor(props) {
         super(props);
 
-        this.popupBodyRef = React.createRef();
-
         // bind callbacks once
-        this._onChangeHandler = this._onChangeHandler.bind(this);
         this._primaryBtnHandler = this._primaryBtnHandler.bind(this);
         this._secondaryBtnHandler = this._secondaryBtnHandler.bind(this);
     }
 
-    componentDidMount(): void {
-        const { inputDefaultValue } = this.props;
-
-        if (inputDefaultValue) {
-            // set the initial value of the prompt input field
-            this.setState({ inputValue: inputDefaultValue.toString() })
+    static getDerivedStateFromProps(props: PopupProps): Partial<PopupState> {
+        if (props.validationErrors != null) {
+            // if there are validation errors, reset the button spinners
+            return {
+                primaryBtnShowSpinner: false,
+                secondaryBtnShowSpinner: false
+            };
         }
+        return null;
     }
 
-    _primaryBtnHandler(e: React.FormEvent<HTMLFormElement>): void {
-        e.preventDefault();
-
-        const inputValue = this.state.inputValue;
-
-        this.setState({ previousInputValue: inputValue });
-
-        if (this.props.validationHandler) {
-            if (this._isValidInput()) {
-                this.props.primaryBtnHandler(inputValue);
-                this.setState({ primaryBtnSpinner: true });
-            }
-        }
-        else {
-            this.props.primaryBtnHandler(inputValue);
-            this.setState({ primaryBtnSpinner: true });
-        }
-    }
-
-    _isValidInput(): boolean {
-        const inputValue = this.state.inputValue;
-
-        const validationErrorMsg = this.props.validationHandler(inputValue);
-
-        if (!validationErrorMsg) {
-            return true;
-        }
-        else {
-            this.setState({ validationErrorMsg: validationErrorMsg });
-            return false;
-        }
+    _primaryBtnHandler(): void {
```

Comment:
missing protected

Reply:
Why should this one be protected, but the others private? Shouldn't it also be private?